### ISSUE TITLE
gpu_people needs to be updated because of nppi splitoff into separate libraries in CUDA 9

### ISSUE THREAD

**Issue by smihael** [PointCloudLibrary/pcl#2388]
nppi is not availiable in CUDA 9 anymore as a single library. See also https://gitlab.kitware.com/cmake/cmake/issues/17310, https://medium.com/@mengjiunchiou/build-opencv-caffe-with-cuda-9-0-on-ubuntu-16-04-b2794a41612d and https://docs.nvidia.com/cuda/npp/index.html.

Due to that trying to build gpu_people results in 
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CUDA_nppi_LIBRARY (ADVANCED)
    linked by target "pcl_gpu_people" in directory ~/Programi/pcl-1.8.1/gpu/people
```
If you ignore this and compile anyway, the binary will not work correctly:

```
development:~/Programi/pcl-1.8.1/$ bin/pcl_people_app 
People tracking App version 0.2
[pcl::gpu::printShortCudaDeviceInfo] : Device 0:  "GeForce GTX 470"  1216Mb, sm_20, 448 cores, Driver/Runtime ver.9.10/9.10
*** stack smashing detected ***: bin/pcl_people_app terminated
Aborted (core dumped)
```

I suspect, that the FindCUDA script from CMake should be updated or dependency for a newer version of CMake should be added. In addition, the following lines of code should possibly be updated.

```
development:~/Programi/pcl-1.8.1/gpu$ grep -rni nppi .
./people/CMakeLists.txt:23:    find_cuda_helper_libs(nppi)
./people/CMakeLists.txt:26:    set(CUDA_npp_LIBRARY ${CUDA_nppc_LIBRARY} ${CUDA_nppi_LIBRARY} ${CUDA_npps_LIBRARY} CACHE STRING "npp library")
./people/src/cuda/nvidia/NCVHaarObjectDetection.cu:1578:    nppStat = nppiStIntegralGetSize_8u32u(NcvSize32u(d_srcImg.width(), d_srcImg.height()), &szTmpBufIntegral, devProp);
./people/src/cuda/nvidia/NCVHaarObjectDetection.cu:1580:    nppStat = nppiStSqrIntegralGetSize_8u64u(NcvSize32u(d_srcImg.width(), d_srcImg.height()), &szTmpBufSqIntegral, devProp);
./people/src/cuda/nvidia/NCVHaarObjectDetection.cu:1587:    nppStat = nppiStIntegral_8u32u_C1R(d_srcImg.ptr(), d_srcImg.pitch(),
./people/src/cuda/nvidia/NCVHaarObjectDetection.cu:1593:    nppStat = nppiStSqrIntegral_8u64u_C1R(d_srcImg.ptr(), d_srcImg.pitch(),
./people/src/cuda/nvidia/NCVHaarObjectDetection.cu:1674:        nppStat = nppiStDecimate_32u_C1R(
./people/src/cuda/nvidia/NCVHaarObjectDetection.cu:1680:        nppStat = nppiStDecimate_64u_C1R(
./people/src/cuda/nvidia/NCVHaarObjectDetection.cu:1691:        nppStat = nppiStRectStdDev_32f_C1R(
./people/src/cuda/nvidia/NPP_staging.cu:374:    ncvStat = nppiStTranspose_32u_C1R((Ncv32u *)Tmp32_1.ptr(), PaddedWidthII32*sizeof(Ncv32u),
./people/src/cuda/nvidia/NPP_staging.cu:383:    ncvStat = nppiStTranspose_32u_C1R((Ncv32u *)Tmp32_1.ptr(), PaddedHeightII32*sizeof(Ncv32u),
./people/src/cuda/nvidia/NPP_staging.cu:436:    ncvStat = nppiStTranspose_32u_C1R(Tmp32_2.ptr(), PaddedWidthII32*sizeof(Ncv32u),
./people/src/cuda/nvidia/NPP_staging.cu:445:    ncvStat = nppiStTranspose_64u_C1R(Tmp64_2.ptr(), PaddedHeightII64*sizeof(Ncv64u),
./people/src/cuda/nvidia/NPP_staging.cu:455:NCVStatus nppiStIntegralGetSize_8u32u(NcvSize32u roiSize, Ncv32u *pBufsize, cudaDeviceProp &devProp)
./people/src/cuda/nvidia/NPP_staging.cu:473:NCVStatus nppiStIntegralGetSize_32f32f(NcvSize32u roiSize, Ncv32u *pBufsize, cudaDeviceProp &devProp)
./people/src/cuda/nvidia/NPP_staging.cu:491:NCVStatus nppiStSqrIntegralGetSize_8u64u(NcvSize32u roiSize, Ncv32u *pBufsize, cudaDeviceProp &devProp)
./people/src/cuda/nvidia/NPP_staging.cu:509:NCVStatus nppiStIntegral_8u32u_C1R(Ncv8u *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:524:NCVStatus nppiStIntegral_32f32f_C1R(Ncv32f *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:539:NCVStatus nppiStSqrIntegral_8u64u_C1R(Ncv8u *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:554:NCVStatus nppiStIntegral_8u32u_C1R_host(Ncv8u *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:586:NCVStatus nppiStIntegral_32f32f_C1R_host(Ncv32f *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:620:NCVStatus nppiStSqrIntegral_8u64u_C1R_host(Ncv8u *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:808:    NCVStatus nppiStDecimate_##bit##typ##_C1R(Ncv##bit##typ *d_src, Ncv32u srcStep, \
./people/src/cuda/nvidia/NPP_staging.cu:819:    NCVStatus nppiStDecimate_##bit##typ##_C1R_host(Ncv##bit##typ *h_src, Ncv32u srcStep, \
./people/src/cuda/nvidia/NPP_staging.cu:949:NCVStatus nppiStRectStdDev_32f_C1R(Ncv32u *d_sum, Ncv32u sumStep,
./people/src/cuda/nvidia/NPP_staging.cu:1006:NCVStatus nppiStRectStdDev_32f_C1R_host(Ncv32u *h_sum, Ncv32u sumStep,
./people/src/cuda/nvidia/NPP_staging.cu:1191:    NCVStatus nppiStTranspose_##bit##typ##_C1R(Ncv##bit##typ *d_src, Ncv32u srcStep, \
./people/src/cuda/nvidia/NPP_staging.cu:1200:    NCVStatus nppiStTranspose_##bit##typ##_C1R_host(Ncv##bit##typ *h_src, Ncv32u srcStep, \
./people/src/cuda/nvidia/NPP_staging.cu:1224:NCVStatus nppiStTranspose_128_C1R(void *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:1231:NCVStatus nppiStTranspose_128_C1R_host(void *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.cu:1719:NCVStatus nppiStFilterRowBorder_32f_C1R(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.cu:1791:NCVStatus nppiStFilterColumnBorder_32f_C1R(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.cu:1962:NCVStatus nppiStGetInterpolationBufferSize(NcvSize32u srcSize,
./people/src/cuda/nvidia/NPP_staging.cu:1967:    status = nppiStVectorWarpGetBufferSize(srcSize, nStep, hpSize);
./people/src/cuda/nvidia/NPP_staging.cu:1972:NCVStatus nppiStInterpolateFrames(const NppStInterpolationState *pState)
./people/src/cuda/nvidia/NPP_staging.cu:2006:        nppiStVectorWarp_PSF2x2_32f_C1 (pState->pFU, 
./people/src/cuda/nvidia/NPP_staging.cu:2016:        nppiStVectorWarp_PSF2x2_32f_C1 (pState->pFV, 
./people/src/cuda/nvidia/NPP_staging.cu:2027:        nppiStVectorWarp_PSF2x2_32f_C1 (pState->pBU, 
./people/src/cuda/nvidia/NPP_staging.cu:2037:        nppiStVectorWarp_PSF2x2_32f_C1 (pState->pBV, 
./people/src/cuda/nvidia/NPP_staging.cu:2232:NCVStatus nppiStVectorWarpGetBufferSize (NcvSize32u srcSize, Ncv32u nSrcStep, Ncv32u *hpSize)
./people/src/cuda/nvidia/NPP_staging.cu:2245:NCVStatus nppiStVectorWarp_PSF1x1_32f_C1(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.cu:2278:NCVStatus nppiStVectorWarp_PSF2x2_32f_C1(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.cu:2514:NCVStatus nppiStResize_32f_C1R(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.hpp:81:/** \defgroup nppi NPPST Image Processing
./people/src/cuda/nvidia/NPP_staging.hpp:144:NCVStatus nppiStGetInterpolationBufferSize(NcvSize32u srcSize,
./people/src/cuda/nvidia/NPP_staging.hpp:157:NCVStatus nppiStInterpolateFrames(const NppStInterpolationState *pState);
./people/src/cuda/nvidia/NPP_staging.hpp:180:NCVStatus nppiStFilterRowBorder_32f_C1R(const Ncv32f *pSrc, 
./people/src/cuda/nvidia/NPP_staging.hpp:214:NCVStatus nppiStFilterColumnBorder_32f_C1R(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.hpp:237:NCVStatus nppiStVectorWarpGetBufferSize(NcvSize32u srcSize,
./people/src/cuda/nvidia/NPP_staging.hpp:263:NCVStatus nppiStVectorWarp_PSF1x1_32f_C1(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.hpp:295:NCVStatus nppiStVectorWarp_PSF2x2_32f_C1(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.hpp:325:NCVStatus nppiStResize_32f_C1R(const Ncv32f *pSrc,
./people/src/cuda/nvidia/NPP_staging.hpp:352:NCVStatus nppiStDecimate_32u_C1R(Ncv32u *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:360: * \see nppiStDecimate_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:363:NCVStatus nppiStDecimate_32s_C1R(Ncv32s *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:371: * \see nppiStDecimate_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:374:NCVStatus nppiStDecimate_32f_C1R(Ncv32f *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:382:* \see nppiStDecimate_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:385:NCVStatus nppiStDecimate_64u_C1R(Ncv64u *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:393: * \see nppiStDecimate_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:396:NCVStatus nppiStDecimate_64s_C1R(Ncv64s *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:404: * \see nppiStDecimate_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:407:NCVStatus nppiStDecimate_64f_C1R(Ncv64f *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:426:NCVStatus nppiStDecimate_32u_C1R_host(Ncv32u *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:433: * \see nppiStDecimate_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:436:NCVStatus nppiStDecimate_32s_C1R_host(Ncv32s *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:443: * \see nppiStDecimate_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:446:NCVStatus nppiStDecimate_32f_C1R_host(Ncv32f *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:453: * \see nppiStDecimate_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:456:NCVStatus nppiStDecimate_64u_C1R_host(Ncv64u *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:463: * \see nppiStDecimate_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:466:NCVStatus nppiStDecimate_64s_C1R_host(Ncv64s *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:473: * \see nppiStDecimate_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:476:NCVStatus nppiStDecimate_64f_C1R_host(Ncv64f *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:498:NCVStatus nppiStRectStdDev_32f_C1R(Ncv32u *d_sum, Ncv32u sumStep,
./people/src/cuda/nvidia/NPP_staging.hpp:521:NCVStatus nppiStRectStdDev_32f_C1R_host(Ncv32u *h_sum, Ncv32u sumStep,
./people/src/cuda/nvidia/NPP_staging.hpp:540:NCVStatus nppiStTranspose_32u_C1R(Ncv32u *d_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:546: * \see nppiStTranspose_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:549:NCVStatus nppiStTranspose_32s_C1R(Ncv32s *d_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:555: * \see nppiStTranspose_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:558:NCVStatus nppiStTranspose_32f_C1R(Ncv32f *d_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:564: * \see nppiStTranspose_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:567:NCVStatus nppiStTranspose_64u_C1R(Ncv64u *d_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:573: * \see nppiStTranspose_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:576:NCVStatus nppiStTranspose_64s_C1R(Ncv64s *d_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:582: * \see nppiStTranspose_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:585:NCVStatus nppiStTranspose_64f_C1R(Ncv64f *d_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:591: * \see nppiStTranspose_32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:594:NCVStatus nppiStTranspose_128_C1R(void *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:610:NCVStatus nppiStTranspose_32u_C1R_host(Ncv32u *h_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:616: * \see nppiStTranspose_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:619:NCVStatus nppiStTranspose_32s_C1R_host(Ncv32s *h_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:625: * \see nppiStTranspose_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:628:NCVStatus nppiStTranspose_32f_C1R_host(Ncv32f *h_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:634: * \see nppiStTranspose_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:637:NCVStatus nppiStTranspose_64u_C1R_host(Ncv64u *h_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:643: * \see nppiStTranspose_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:646:NCVStatus nppiStTranspose_64s_C1R_host(Ncv64s *h_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:652: * \see nppiStTranspose_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:655:NCVStatus nppiStTranspose_64f_C1R_host(Ncv64f *h_src, Ncv32u srcStride,
./people/src/cuda/nvidia/NPP_staging.hpp:661: * \see nppiStTranspose_32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:664:NCVStatus nppiStTranspose_128_C1R_host(void *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:678:NCVStatus nppiStIntegralGetSize_8u32u(NcvSize32u roiSize, Ncv32u *pBufsize, cudaDeviceProp &devProp);
./people/src/cuda/nvidia/NPP_staging.hpp:683: * \see nppiStIntegralGetSize_8u32u
./people/src/cuda/nvidia/NPP_staging.hpp:686:NCVStatus nppiStIntegralGetSize_32f32f(NcvSize32u roiSize, Ncv32u *pBufsize, cudaDeviceProp &devProp);
./people/src/cuda/nvidia/NPP_staging.hpp:704:NCVStatus nppiStIntegral_8u32u_C1R(Ncv8u *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:711: * \see nppiStIntegral_8u32u_C1R
./people/src/cuda/nvidia/NPP_staging.hpp:714:NCVStatus nppiStIntegral_32f32f_C1R(Ncv32f *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:731:NCVStatus nppiStIntegral_8u32u_C1R_host(Ncv8u *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:737: * \see nppiStIntegral_8u32u_C1R_host
./people/src/cuda/nvidia/NPP_staging.hpp:740:NCVStatus nppiStIntegral_32f32f_C1R_host(Ncv32f *h_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:754:NCVStatus nppiStSqrIntegralGetSize_8u64u(NcvSize32u roiSize, Ncv32u *pBufsize, cudaDeviceProp &devProp);
./people/src/cuda/nvidia/NPP_staging.hpp:772:NCVStatus nppiStSqrIntegral_8u64u_C1R(Ncv8u *d_src, Ncv32u srcStep,
./people/src/cuda/nvidia/NPP_staging.hpp:789:NCVStatus nppiStSqrIntegral_8u64u_C1R_host(Ncv8u *h_src, Ncv32u srcStep,
./people/src/cuda/utils.cu:102:  NppiSize sz;
./people/src/cuda/utils.cu:105:  nppSafeCall( nppiSet_8u_C1R( 0, mask, (int)mask.step(), sz) );
./people/src/cuda/utils.cu:122:  NppiSize sz;
./people/src/cuda/utils.cu:126:  NppiSize ksz;
./people/src/cuda/utils.cu:130:  NppiPoint anchor;
./people/src/cuda/utils.cu:135:  nppSafeCall( nppiDilate_8u_C1R(src.ptr(ANCH_Y) + ANCH_X, (int)src.step(),
```

**Comment by SergioRAgostinho** [comment]
Sorry for the delay. I believe that was fixed in #2181. Did you try compiling the current master branch?

**Comment by SergioRAgostinho** [comment]
Closing due to inactivity.
